src/connection.rs

```diff
@@ -180,7 +180,6 @@ pub enum ConnectionState {
 impl From<i64> for ConnectionState {
     fn from(state: i64) -> Self {
         match state {
-            0 => ConnectionState::Unknown,
```

Comment:
I think we should keep `0` as its a valid state and change the default `_` state to throw some kind of error as it should never be hit.